src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo.cs

```diff
@@ -319,33 +320,22 @@ private static int BinarySearch(ImmutableArray<Node> nodes, string name)
 
         #region Construction
 
-        // Cache the symbol tree infos for assembly symbols that share the same underlying metadata.
-        // Generating symbol trees for metadata can be expensive (in large metadata cases).  And it's
-        // common for us to have many threads to want to search the same metadata simultaneously.
-        // As such, we want to only allow one thread to produce the tree for some piece of metadata
-        // at a time.  
-        //
-        // AsyncLazy would normally be an ok choice here.  However, in the case where all clients
-        // cancel their request, we don't want ot keep the AsyncLazy around.  It may capture a lot
-        // of immutable state (like a Solution) that we don't want kept around indefinitely.  So we
-        // only cache results (the symbol tree infos) if they successfully compute to completion.
```

Comment:
this comment is very relevant.  we switched off of AsyncLazy because we woudl have AsyncLazys holding onto SOlution instances.  However, in the new code, we do not hold onto solution instances.  Instead, we hold onto a Workspace+SolutionKey.  This is possible as the SOlution was only used to acquire an IPersistentStorage instance associated with that solution.  But we don't need a full Solution object for that anymore with the current IPersistentStorage apis.

Note: if holding onto a workspace here is problematic, we could always key this first off of Workspace.